File: Milkweed/Milkweed.cpp
#include "Milkweed.h"

Milkweed::Milkweed(World *world, int x, int y)
    : Plant(MILKWEED_STRENGTH, MILKWEED_SYMBOL, MILKWEED_COLOR, world, x, y) {}

void Milkweed::action() {
  Plant::action();
  Plant::action();
  Plant::action();
}

void Milkweed::collision(Organism* organism) { }

std::string Milkweed::getType() const { return "Milkweed"; }

Organism* Milkweed::createChild(int x, int y, int parent1Id, int parent2Id) {
  return new Milkweed(this->getWorld(), x, y);
}

File: src/Milkweed/Milkweed.java
package Milkweed;

import Organism.*;
import World.World;
import Plant.Plant;
import Field.Field;
import Settings.Settings;

public class Milkweed extends Plant {

    public Milkweed(World world, Field field) {
        super(new OrganismData(world.getNextOrganismId(), Settings.MILKWEED_INITIATIVE, Settings.MILKWEED_STRENGTH, Settings.MILKWEED_IMAGE, -1, -1), world, field);
    }

    @Override
    public void action() {
        super.action();
        super.action();
        super.action();
    }

    @Override
    public void collision(Organism attacker) {
        super.collision(attacker);
    }

    @Override
    public String getType() {
        return "Milkweed";
    }

    @Override
    public void reproduce(Field field) {
        Organism milkweed = new Milkweed(this.world, field);
        this.world.addOrganism(milkweed);
        this.world.addLog(this.getType() + " reproduced (" + field.getNumber() + ")");
    }
}
